lib/encoding/EncodeInstruction.ml

```diff
@@ -0,0 +1,200 @@
+open Isa
+open PAst
+open Integers
+open EncodedFile
+
+let encode_opcode (pos, v) inst =
+  let opcode_size = 4 in
+  let c =
+    match inst with
+    | And _ | Or _ | Nor _ | Xor _ | Add _ | Sub _ | Mul _ | Div _ -> 0
+    | ShiftLeftLogical _ -> 1
+    | ShiftRightArith _ -> 2
+    | ShiftRightLogical _ -> 3
+    | Load _ -> 4
+    | LoadImmediateAdd _ -> 5
+    | Store _ -> 6
+    | Jmp _ -> 7
+    | JmpCond _ -> 8
+    | JmpImmediate _ -> 9
+    | JmpImmediateCond _ -> 10
+  in
+  (pos + opcode_size, Word.add_at v (Int32.of_int c) pos)
+
+let encode_reg (pos, v) reg =
+  let reg_size = 5 in
+  let c =
+    match reg with
+    | R0 -> 0
+    | R1 -> 1
+    | R2 -> 2
+    | R3 -> 3
+    | R4 -> 4
+    | R5 -> 5
+    | R6 -> 6
+    | R7 -> 7
+    | R8 -> 8
+    | R9 -> 9
+    | R10 -> 10
+    | R11 -> 11
+    | R12 -> 12
+    | R13 -> 13
+    | R14 -> 14
+    | R15 -> 15
+    | R16 -> 16
+    | R17 -> 17
+    | R18 -> 18
+    | R19 -> 19
+    | R20 -> 20
+    | R21 -> 21
+    | R22 -> 22
+    | R23 -> 23
+    | R24 -> 24
+    | R25 -> 25
+    | R26 -> 26
+    | R27 -> 27
+    | ROut -> 28
+    | SP -> 29
+    | FP -> 30
+    | PrivateReg -> 31
+  in
+  (pos + reg_size, Word.add_at v (Int32.of_int c) pos)
+
+let encode_alucode (pos, v) inst =
+  let alu_size = 5 in
+  let c =
+    match inst with
+    | And _ -> 0
+    | Or _ -> 1
+    | Nor _ -> 2
+    | Xor _ -> 3
+    | Add _ -> 4
+    | Sub _ -> 5
+    | Mul _ -> 6
+    | Div _ -> 7
+    | _ -> failwith "Instruction not in the ALU !"
+  in
+  (pos + alu_size, Word.add_at v (Int32.of_int c) pos)
+
+let encode_imm (pos, v) imm16 =
+  let imm_size = 16 in
+  let c = UInt16.encode imm16 in
+  (pos + imm_size, Word.add_at v c pos)
+
+let encode_loadmode (pos, v) mode =
+  let loadmode_size = 1 in
+  let c = match mode with LowHalf -> Int32.zero | HighHalf -> Int32.one in
```

Comment:
C'est l'inverse d'après la doc.
LowHalf -> **1**
HighHalf -> **0**

Reply:
Fixed in 72d69c40ea0b504a8e8e094c10db8f00469490e7.